### PR TITLE
Fixed index column stickiness mobile

### LINKED ISSUE #4155: Adapt Index first column Stickiness on Mobile
# Current behavior

The Index first column is Sticky when horizontally scrolling the columns

image

# Desired Behavior

We want to adapt the first column stickiness **on Mobile** as it prevents good content readability. We want to keep only the companies Favicon and the checkboxes as we scroll to the right

image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=23796-35984&mode=design&t=0qrJcy4ODeblizYh-11

### PR DESCRIPTION
**Pull request by jeet1desai**
Fixes #4155 

https://github.com/twentyhq/twenty/assets/52026385/928e9f38-d49a-46be-89ea-045f8ae6c149

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-table/components/RecordTableHeaderCell.tsx [hunk 1]
```diff
@@ -73,6 +74,7 @@ const StyledHeaderIcon = styled.div`
   margin-bottom: ${({ theme }) => theme.spacing(1)};
   margin-right: ${({ theme }) => theme.spacing(1)};
   margin-top: ${({ theme }) => theme.spacing(1)};
+  margin-left: ${({ theme }) => theme.spacing(1.5)};
```
[the remark this answers is not in the document]
Comment (thaisguigon) [reply packages/twenty-front/src/modules/object-record/record-table/components/RecordTableHeaderCell.tsx:77]: Also, we should disable resizing the label identifier column when it's in mobile "sticky" mode.

We could disable it like this same file (`RecordTableHeaderCell`):

```typescript
  const isMobile = useIsMobile();
  const scrollLeft = useRecoilValue(scrollLeftState);

  const disableColumnResize =
    column.isLabelIdentifier && isMobile && scrollLeft > 0;

  // ...

  {!disableColumnResize && (
    <StyledResizeHandler
      className="cursor-col-resize"
      role="separator"
      onPointerDown={() => {
        setResizedFieldKey(column.fieldMetadataId);
      }}
    />
  )}
```

**Review by thaisguigon — CHANGES_REQUESTED** [review]
Hi @jeet1desai, thank you for your contribution!

I pushed an additional commit to remove `RecordTableFirstColumnScrollEffect` and simplify the "frozen columns" logic in `RecordTable`.

I also left some other comments. Nice work overall!

**Review by thaisguigon — DISMISSED** [review]
Hi @jeet1desai, thank you for addressing the comments!
I see you resolved most of them, the only one missing is https://github.com/twentyhq/twenty/pull/4206#discussion_r1507703211, could you look into it?
This is currently breaking the typings!

**Review by charlesBochet — APPROVED** [review]
Thank you @jeet1desai 
and thanks @thaisguigon for the review!

**Pull request merged**